source/opt/loop_descriptor.h

```diff
@@ -68,14 +68,58 @@ class Loop {
   // OpLoopMerge instruction.
   inline BasicBlock* GetHeaderBlock() { return loop_header_; }
   inline const BasicBlock* GetHeaderBlock() const { return loop_header_; }
+  inline void SetHeaderBlock(BasicBlock* header) { loop_header_ = header; }
+
+  // Returns true if the loop can be considered as structured according to
+  // SPIR-V rules:
+  //  - Has a merge block;
+  //  - Has a continue block.
+  inline bool IsStructuredCompliant() const {
+    return GetLatchBlock() && GetMergeBlock();
+  }
+
+  // Returns true if the loop is structured according to SPIR-V rules:
+  //  - Has a merge block;
+  //  - Has a continue block;
+  //  - Has an OpLoopMerge instruction.
+  inline bool IsStructured() const {
+    return IsStructuredCompliant() && GetHeaderBlock()->GetLoopMergeInst();
+  }
+
+  // Updates the OpLoopMerge instruction to reflect the current state of the
+  // loop.
+  inline void UpdateLoopMergeInst() {
+    assert(IsStructured() && "The loop is not structured");
+    ir::Instruction* merge_inst = GetHeaderBlock()->GetLoopMergeInst();
+    merge_inst->SetInOperand(0, {GetMergeBlock()->id()});
+    merge_inst->SetInOperand(1, {GetLatchBlock()->id()});
+  }
 
   // Returns the latch basic block (basic block that holds the back-edge).
+  // These functions return nullptr if the loop is not structured (i.e. if it
+  // has more than one backedge).
   inline BasicBlock* GetLatchBlock() { return loop_continue_; }
   inline const BasicBlock* GetLatchBlock() const { return loop_continue_; }
-
-  // Returns the BasicBlock which marks the end of the loop.
+  // Sets |latch| as the loop unique continue block. A continue block must have
```

Comment:
The latch and the continue blocks can be different blocks in structured control flow.

Reply:
Yes, I realized that now... I'll remove the update of the continue block.